test/client.py

```diff
@@ -78,6 +80,53 @@ def test_sync_import(self):
         fi.run()
         self.assertIsNotNone(fi.id)
 
+    def test_import_jobs_length(self):
+        import_id = None
+        manager = FileImportManager(self.client)
+        fi = FileImport(IMPORT_FILE, self.client)
+        all_imports = manager.all()
+        self.assertEqual(len(all_imports), 1)
+        import_id = all_imports[0].id
+        self.assertIsNotNone(import_id)
+
+    def test_updated_job_id(self):
+        fi = FileImport(IMPORT_FILE, self.client)
+        fi.run()
+        self.assertEqual(fi.success, True)
+        initial_id = fi.id
+        has_state = True
+        try:
+            state = fi.state
+        except AttributeError:
+            has_state = False
+        self.assertEqual(has_state, False)
+        fi.update()
+        self.assertEqual(fi.state, 'pending')
+        final_id = fi.id
+        self.assertEqual(initial_id, final_id)
+
+
+
+class ImportErrorTest(unittest.TestCase):
+    def setUp(self):
+        self.client = APIKeyAuthClient(API_KEY, USER)
+        
+    def test_error_handling(self):
+        fi = FileImport("test/fake.html", self.client)
+        fi.run()
+        self.assertEqual(fi.success, True)
+        fi.update()
+        count = 0
+        while fi.state != 'failure':
+            time.sleep(5)
+            fi.update()
+            if count == 10:
```

Comment:
I'd move this to the beginning of the while loop. besides, you're not incrementing `c` as far as I can see